### DISCUSSION TITLE
how to start node js project in a EC2 AWS instance

### DISCUSSION THREAD

**Discussion by chandnie** [community/community#20667] [Other Feature Feedback, Questions, & Ideas] [answered]
I am completely new to node js, I am trying to set it up in EC2 AWS. I tried using the command "npm start" but nothing has happened

below is my package.json

```
{
  "name": "sample",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "start": "node ./bin/www"
  },
  "dependencies": {
    "cookie-parser": "~1.4.4",
    "debug": "~2.6.9",
    "ejs": "~2.6.1",
    "express": "~4.16.1",
    "http-errors": "~1.6.3",
    "morgan": "~1.9.1",
    "pm2": "^5.1.2"
  }
}bouldercef$ npm start
```
after running npm start the below lines appear, but not able to see the site

**Comment by npmInicola90** [comment]
https://ourcodeworld.com/articles/read/977/how-to-deploy-a-node-js-application-on-aws-ec2-server, this might help :)

**Comment by magikadev** [comment]
Is the project working locally?

For deploying the site with node, you should need to use nginx.

For running backend or api, you just need to point the .js file.
The scripts should be `"start": "node ./bin/www/app.js"`

**Comment by chiragmehta900** [comment, marked as answer]
If your site is running properly in your machine, and if you have node/npm installed in your EC2 instance and you are not receiving any errors, looks like your project is running without issues.

So the problem can be that you are not connect to your site using HTTP.

Take a look on the security group attached with your instance to check if you allow users to connect on HTTP (80) or HTTPS (443).

To more information of how to make these ports available for your users check [How do I allow my users to connect on HTTP (80) or HTTPS (443)?](https://aws.amazon.com/premiumsupport/knowledge-center/connect-http-https-ec2/)
